src/cmd-buildextend-gcp

```diff
@@ -7,94 +7,138 @@ extending the meta.json with GCP image name.
-gcp_tarball_name = f'{gcp_name_version}-gcp.tar.gz'
-
-# Setup the tempdir
-tmpdir = 'tmp/buildpost-gcp'
-if os.path.isdir(tmpdir):
-    shutil.rmtree(tmpdir)
-os.mkdir(tmpdir)
-tmpdir = os.path.abspath(tmpdir)
-
-def generate_gcp_tar():
-    """ Set the Ignition ID to GCP/GCE and create a tarball for upload """
-    plat_id = 'gcp'
-    if args.gce:
-        plat_id = 'gce'
-    img_qemu = os.path.join(builddir, buildmeta['images']['qemu']['path'])
-    tmp_img_gcp = os.path.join(tmpdir, (gcp_name_version + '.qcow2'))
-    tmp_img_gcp_raw = os.path.join(tmpdir, 'disk.raw')
-    tmp_img_gcp_tar = os.path.join(tmpdir, gcp_tarball_name)
-    run_verbose([f"{cosa_dir}/gf-platformid", img_qemu, tmp_img_gcp, plat_id])
-    run_verbose(['qemu-img', 'convert', '-f', 'qcow2', '-O', 'raw',
-                 tmp_img_gcp, tmp_img_gcp_raw])
-    run_verbose(['tar', '-C', f"{tmpdir}", '-Sczf', f"{tmp_img_gcp_tar}",
-                 f"disk.raw"])
-    os.unlink(tmp_img_gcp)
-    os.unlink(tmp_img_gcp_raw)
-    return tmp_img_gcp_tar
-
-def run_ore():
-    """ Execute ore to upload the tarball and register the image """
-    build_tarball = f"{builddir}/{gcp_tarball_name}"
-    tmp_img_gcp_tarball = generate_gcp_tar()
+        sha256sum_file)
+
+from cosalib.build import _Build
+
+# Temporary directory used for this command
+TMPDIR = os.path.abspath('tmp/buildpost-gcp')
+
+
+class Build(_Build):
+    """
+    GCP implementation of Build.
+    """
+
+    def _build_artifacts(self, *args, **kwargs):
+        """
+        Implements the building of artifacts. Walk the build root and
+        prepare a list of files in it.
+
+        :param args: All non-keyword arguments
+        :type args: list
+        :param kwargs: All keyword arguments
+        :type kwargs: dict
+        """
+        # Setup the tempdir
+        if os.path.isdir(TMPDIR):
+            shutil.rmtree(TMPDIR)
+        os.mkdir(TMPDIR)
+
+        # Name the base build and tarball name
+        base_name = self.meta['name']
+        if args[0].name_suffix:
+            gcp_nv = f'{base_name}-{args[0].name_suffix}-{args[0].build}'
+        else:
+            gcp_nv = f'{base_name}-{args[0].build}'
+        # Used in uploading
+        self.gcp_tarball_name = f'{gcp_nv}-gcp.tar.gz'
+
+        plat_id = 'gcp'
+        if args[0].gce:
+            plat_id = 'gce'
+        img_qemu = os.path.join(
+            self.build_root, self.meta['images']['qemu']['path'])
+        tmp_img_gcp = os.path.join(TMPDIR, (gcp_nv + '.qcow2'))
+        tmp_img_gcp_raw = os.path.join(TMPDIR, 'disk.raw')
+        tmp_img_gcp_tar = os.path.join(TMPDIR, self.gcp_tarball_name)
+        run_verbose([f"{cosa_dir}/gf-platformid",
+                     img_qemu, tmp_img_gcp, plat_id])
+        run_verbose(['qemu-img', 'convert', '-f', 'qcow2', '-O', 'raw',
+                    tmp_img_gcp, tmp_img_gcp_raw])
+        run_verbose(['tar', '-C', f"{TMPDIR}", '-Sczf', f"{tmp_img_gcp_tar}",
```

Comment:
The GCP docs say:

> The compressed file must be a .tar.gz file that uses gzip compression and the --format=oldgnu option for the tar utility (manual).

Does it work without that `--format` arg?

Reply:
I'll add this in, but I believe @darkmuggle was able to upload via the command.